### DISCUSSION TITLE
How do I change the default keybindings?

### DISCUSSION THREAD

**Discussion by txtyash** [LazyVim/LazyVim#1094] [Q&A] [answered]
I want to change the default `C-s` to do something else.
This didn't work for me(keymaps.lua):
```lua
map({ "i", "v", "n", "s" }, { "<C-s>d", "<C-s><C-d>" }, "vim.diagnostic.open_float", { desc = "diagnostics" })
```
Should I change `require("lazyvim.config").setup({})`?

**Comment by dpetka2001** [comment]
That's because `vim.diagnostic` is considered LSP keymap and these you have to change a bit differently. See [docs](https://www.lazyvim.org/plugins/lsp#%EF%B8%8F-customizing-lsp-keymaps).

**Reply by txtyash** [reply, marked as answer]
This works:
```lua
return {
  -- LSP keymaps
  {
    "neovim/nvim-lspconfig",
    init = function()
      local keys = require("lazyvim.plugins.lsp.keymaps").get()
      keys[#keys + 1] = { "jk", vim.diagnostic.open_float, mode = "n" }
      keys[#keys + 1] = { "kj", vim.diagnostic.open_float, mode = "n" }
    end,
  },
}
```

**Comment by jyuan0** [comment]
Shouldn't the `lhs` also be a string, rather than a table?

**Reply by dpetka2001** [reply]
What do you mean? The `lhs` is a string. In `keys[#keys + 1] = { "K", "<cmd>echo 'hello'<cr>" }` the `lhs` is "K" and `rhs` is the command expression to execute.

**Reply by jyuan0** [reply]
Sorry, I meant the `{ "<C-s>d", "<C-s><C-d>" }` part. Shouldn't that be a string?

**Reply by dpetka2001** [reply]
Yes, I believe it should. Though it still shouldn't work in this case as he wants to create custom LSP mappings. He should follow the way it's shown in the docs for LSP mappings.

**Reply by jyuan0** [reply]
Sorry, didn't mean to cause any confusion or anything. Thanks for the reply.

**Reply by dpetka2001** [reply]
No confusion at all and no need for apologies. I also have asked so many questions and people helped me understand things.

**Reply by dpetka2001** [reply]
But I believe he should also disable the mapping of `<C-s>` in `keymaps.lua` so that the keybinding can be available to assign in LSP. Because I'm not sure which is the order in terms of loading keymaps (first `keymaps.lua` and then LSP mappings or the other way around) or even if one case overwrites the other if both have the same keybinding.

**Reply by txtyash** [reply]
Here you go @jyuan0 
```lua
local function map(modes, lhs, rhs, opts)
  if type(modes) == "string" then
    modes = { modes }
  end
  local keys = require("lazy.core.handler").handlers.keys
  for _, mode in pairs(modes) do
    if not keys.active[keys.parse({ lhs, mode = mode }).id] then
      opts = opts or {}
      opts.silent = opts.silent ~= false
      vim.keymap.set(mode, lhs, rhs, opts)
    end
  end
end
```

**Reply by dpetka2001** [reply]
In terms of that function then yes. I was thinking in terms of `local map = vim.keymap.set` (since that's what I use in my `keymaps.lua`), thus thinking that `lhs` should have been string.

**Reply by jyuan0** [reply]
> Here you go @jyuan0

Thanks, although I wasn't asking for that. I was assuming `map` was `vim.keymap.set`.

**Reply by folke** [reply]
@zim0369 there's no need for you to use that keymap function. That's only there for LazyVim, so LV doesn't override any custom plugin keymaps a user added that clash with the defaults.

LSP keymaps are loaded every time an lsp attaches to a buffer.
